src/Evolution/Systems/ScalarWave/Characteristics.cpp

```diff
@@ -0,0 +1,193 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/ScalarWave/Characteristics.hpp"
+
+#include <algorithm>
```

Comment:
I'll let you take a pass at getting the includes right here before I go through them too...